packages/vanilla-extract/src/index.ts

```diff
@@ -9,34 +9,59 @@ import { precomputeValues } from '@capsizecss/core';
 import { ComputedValues, CreateStyleObjectParameters } from './types';
 import { capsizeStyle, capsizeVars } from './capsize.css';
 
-interface MediaQueries {
-  '@media': Record<string, CreateStyleObjectParameters>;
+interface ConditionalQueries {
+  '@media'?: Record<string, CreateStyleObjectParameters>;
+  '@container'?: Record<string, CreateStyleObjectParameters>;
 }
 
+const generateQueryVars = ({
+  queryType,
+  queries,
+}: {
+  queryType: '@media' | '@container';
+  queries: Record<string, any>;
+}) => {
+  const queryVars: StyleRule[typeof queryType] = {};
+
+  Object.entries(queries).forEach(([query, value]) => {
+    const builtValues =
+      'capHeightTrim' in value
+        ? (value as ComputedValues)
+        : precomputeValues(value);
+
+    queryVars[query] = { vars: assignVars(capsizeVars, builtValues) };
+  });
+
+  return queryVars;
```

Comment:
Can i suggest renaming `queryVars` to `queryRule` given it is generating the ruleset:

```suggestion
  const queryRule: StyleRule[typeof queryType] = {};

  Object.entries(queries).forEach(([query, value]) => {
    const builtValues =
      'capHeightTrim' in value
        ? (value as ComputedValues)
        : precomputeValues(value);

    queryRule[query] = { vars: assignVars(capsizeVars, builtValues) };
  });

  return queryRule;
```

That would go for the function too, `generateQueryVars` to `generateQueryRule`.

---

(I noticed this is likely extending something that was my fault, in `createVanillaStyle` we build up a [`vars`] variable, which is actually a "styleRule". You can update that for bonus points 😄)

[`vars`]: https://github.com/seek-oss/capsize/pull/147/files#diff-f8c0e1ab6c8b75aefdd908e0db5f9fc6bd7ed99dcabe07ec792217e9c477cfc5R47